prow/crier/controller.go

```diff
@@ -199,9 +208,14 @@ func (r *reconciler) reconcile(ctx context.Context, log *logrus.Entry, req recon
 		return requeue, nil
 	}
 
-	log.Info("Reported job(s), now will update pj(s)")
+	log.Infof("Reported job(s), now will update %d pj(s)", len(pjs))
 	for _, pjob := range pjs {
-		if err := r.updateReportStateWithRetries(ctx, pjob, log); err != nil {
+		// Spend up to 10 mins attempting to update report state since the behavior
+		// is poor if we fail. This long of a delay is needed for when the API
+		// server is temporarily unavailable.
+		updateContext, updateCancel := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
Similarly, this is unlikely to do anything because the real cap is the `retry.DefaultBackoff` so I think this will only make a difference if a single UPDATE call takes 10 minutes?

Reply:
Not clear to me whether `updateReportStateWithRetries()` spawns off a go-routine that can take however long it wants.

If it does then it seems like just having a single `ctx, cancel := context.WithTimeout(parent, 10 * minute) // or hour or whatever` seems sufficient to me.

AKA what we really want to avoid is a situation where a single reconcile loop takes 7 hours because of some oddity -- in that situation we can just error out and fix it up on the next reconcile loop.

Reply:
> so I think this will only make a difference if a single UPDATE call takes 10 minutes?

I was thinking this is what would happen if the API server was unavailable due to a master upgrade or reconfiguration, but maybe the requests would just time out?

> Not clear to me whether `updateReportStateWithRetries()` spawns off a go-routine that can take however long it wants.

I don't see us doing that in our code, but I'm not sure if the controller runtime client spawns go-routines to GET or PATCH. That seems possible, but a bit unlikely.
I agree that a single timeout context for updating the report state across all PJs seems sufficient here.